### DISCUSSION TITLE
Decide on a web framework: Wordpress or Django?

### DISCUSSION THREAD

**Discussion by janbaykara** [commonknowledge/leftbookclub#21] [Project Decisions] [answered]
Leaning towards Django for developer efficiency, Wordpress for admin agency.

### Website requirements

- [ ] #9
  - explain what the left book club is
  - [ ] #13
  - [ ] #12
  - blog series would be an easy win
- [ ] #10
  - some way for editors to manage what product plans show up and how? 
  - this ends up at Stripe checkout
  - some way of creating a user and relating it to a Stripe customer
  - ideally this would involve SSO to Twitter, Facebook, Google.
- [ ] #6
- [ ] #11
  - When you buy a gift, we generate a coupon for a specific plan. This coupon : plan relationship should probably be stored in the database, so that when you enter the coupon, it can set up the prepaid subscription
    - Either [generate a checkout link](https://stripe.com/docs/billing/subscriptions/coupons#using-coupons-in-checkout) for you to use, which will require payment details after the end of the prepaid period (maybe this is the MVP?)
    - ... or we avoid checkout, request your customer details (name, shipping address) via a web form and update Stripe via API, avoiding your billing details being put in. (Maybe this is a v2 if we find people don't want to put in billing details?)
- [ ] #4
  - pull in Stripe subscriptions that relate to a user
  - list the backlog of books with a buy button, that funnel visitors into the buy flow (Shopify Storefront API or Stripe Checkout)
- [ ] Logged in users can manage their Stripe subscription
  - [ ] #14
    - We can programmatically un-include the 'cancel' option on the customer portal, so that this can be handled by our own 'retention strategy' web page https://stripe.com/docs/api/customer_portal/configuration#portal_configuration_object-features-subscription_cancel 
  - [ ] #15

### My hot take

I think there is an argument to be made for Wordpress, something around admin agency and customisability? For example, making landing pages for advertising is a lot easier in Wordpress.

However there are at least four user flows I'm concerned about implementing with Wordpress:

- Sign up a user account, generate a Stripe subscription and link the two together
  - Related: editors can somehow arrange/edit copy for the subscription offers
  - Related: allow SSO via Twitter, Facebook, Google for signups?
- Redeem a gift coupon, generate a Stripe subcription and link the two together
- When a user has an active Stripe account, allow them to click the 'buy book' option
  - Related: pull Shopify / Stripe products in via API and list them with checkout buttons
- Visit your customer profile and
  1. Access the Stripe customer portal
  2. Visit a custom cancellation page that would then interact with the Stripe API

The user data requirements of this system and these user flows makes me lean towards Django, with something like Wagtail for page editing.

**Comment by janbaykara** [comment]
@conatus @chrisdevereux @Moggach pile in!

**Comment by conatus** [comment, marked as answer]
Looking at the user flows you are concerned at doing in WordPress, you _could_ do all of these in WordPress.

But I feel it would be bending WordPress quite far to do so, whereas in Django, this seems much more straightforward. So I am going to say, go Django/Wagtail here. 

It means for a slightly less good editing experience in terms of being able to self-serve, but isn't so bad, given the other bits of the project here are more important and level of full page editability is not likely to be high.

**Reply by janbaykara** [reply]
Going to take this and run with it.
